samples/apps/nobuiltins/nobuiltins.cpp

```diff
@@ -227,6 +235,50 @@ namespace nobuiltins
           ccf::endpoints::ExecuteOutsideConsensus::Locally)
         .set_auto_schema<void, TransactionIDResponse>()
         .install();
+
+      auto get_time = [this](auto& ctx, nlohmann::json&&) {
+        ::timespec time;
+        ccf::ApiResult result = get_untrusted_host_time_v1(time);
+        if (result != ccf::ApiResult::OK)
+        {
+          return ccf::make_error(
+            HTTP_STATUS_INTERNAL_SERVER_ERROR,
+            ccf::errors::InternalError,
+            fmt::format(
+              "Unable to get time: {}", ccf::api_result_to_str(result)));
+        }
+
+        std::tm calendar_time;
+        gmtime_r(&time.tv_sec, &calendar_time);
+
+        // 20 characters for a (timezoneless) ISO 8601 datetime, plus a
+        // terminating null
+        constexpr size_t buf_size = 21;
+        char buf[buf_size];
+        if (strftime(buf, buf_size, "%Y-%m-%dT%H:%M:%S", &calendar_time) == 0)
```

Comment:
consider adding milliseconds? We may not need it now but adding it later would be a breaking change

Reply:
I add microseconds below (there's no specifier for milliseconds with strftime, and `calendar_time` only goes down to second precision). Also we don't care about breaking changes to this endpoint - its just a sample endpoint so we can tweak the sample formatting here if we want to recommend something else. The API guarantee is on the `get_untrusted_host_time_v1` API, which returns a `timespec`.